ledger-core/virtual/integration/staterequest_test.go

```diff
@@ -64,14 +64,23 @@ func TestVirtual_VStateRequest_WithoutBody(t *testing.T) {
 		server.SendMessage(ctx, msg)
 	}
 
-	msg := makeVStateRequestEvent(server.GetPulse().PulseNumber, objectRef, 0)
+	server.WaitActiveThenIdleConveyor()
+
+	server.IncrementPulse(ctx)
+
+	server.WaitActiveThenIdleConveyor()
+
+	// skip StateReport from Pulse
+	<-reportChan
+
+	msg := makeVStateRequestEvent(server.GetPulse().PrevPulseNumber, objectRef, 0)
 	server.SendMessage(ctx, msg)
 
 	select {
 	case data := <-reportChan:
 		assert.Equal(t, &payload.VStateReport{
 			Status:           payload.Ready,
-			AsOf:             server.GetPulse().PulseNumber,
+			AsOf:             server.GetPulse().PrevPulseNumber,
```

Comment:
hmm, no, please extract previous pulse from pulsegenerator and use it here

Reply:
for consistency with code

Reply:
created helper server.GetPrevPulse that return prev pulse